src/service/xhr-impl.js

```diff
@@ -139,6 +139,9 @@ export class Xhr {
             ` ${ALLOW_SOURCE_ORIGIN_HEADER} header`);
       }
       return response;
+    }).catch(reason => {
+      user.assert(false, 'Fetch failed %s: %s', input,
```

Comment:
@dvoytenko @jridgewell if i'm understanding correctly the original intent of the catch was just to throw additional info on the error message so it should be fine to move this here at the bottom. (fixes the type error of compiler thinking the catch should return something which the next `then` handler should receive a its paramter)